GLAA.Services/PublicRegister/PublicRegisterViewModelBuilder.cs

```diff
@@ -0,0 +1,141 @@
+            PublicRegisterSearchViewModel publicRegisterSearchViewModel)
+        {
+            var ukCountryNames = UkCountries.Select(x => x.Text);
+
+            var licences = licenceRepository.GetAllLicences()
+                .Where(x => GetLatestStatus(x).Status.ShowInPublicRegister
+                            && (string.IsNullOrWhiteSpace(publicRegisterSearchViewModel.BusinessName) ||
+                                x.BusinessName.Contains(publicRegisterSearchViewModel.BusinessName)));
+
+            switch (publicRegisterSearchViewModel.SupplierWho)
+            {
+                case "supply":
+                    if (publicRegisterSearchViewModel.CountriesSelected.Any(x => x == "UK"))
+                        licences = licences.Where(x =>
+                            ukCountryNames.Any(y => x.OperatingCountries.Any(z => z.Country.Name == y)));
+                    else if (publicRegisterSearchViewModel.CountriesSelected.Any(x => x == "Outside UK"))
+                        licences = licences.Where(x =>
+                            ukCountryNames.Any(y => x.OperatingCountries.All(z => z.Country.Name != y)));
+                    else
+                        licences = licences.Where(x => x.OperatingCountries.Any(y =>
+                            publicRegisterSearchViewModel.CountriesSelected.Any(z => y.Country.Name.Contains(z))));
+
+                    break;
+                case "arelocated":
+                    if (publicRegisterSearchViewModel.CountriesSelected.Any(x => x == "UK"))
+                        licences = licences.Where(x => x.Address != null && ukCountryNames.Any(y => x.Address.Country.Contains(y)));
+                    else if (publicRegisterSearchViewModel.CountriesSelected.Any(x => x == "Outside UK"))
+                        licences = licences.Where(x => x.Address != null && ukCountryNames.Any(y => x.Address.NonUK));
+                    else
+                        licences = licences.Where(x => x.Address != null && publicRegisterSearchViewModel.CountriesSelected.Any(y => y == x.Address.Country));
+
+                    break;
+                default:
+                    throw new InvalidEnumArgumentException(nameof(publicRegisterSearchViewModel.SupplierWho));
+            }
+
+            return new PublicRegisterLicenceListViewModel
+            {
+                Title = "Public Register",
+                Licences = licences.Distinct().Select(BuildSummary),
+                PublicRegisterSearchViewModel = publicRegisterSearchViewModel
+            };
+        }
+
+        public PublicRegisterLicenceSummaryViewModel BuildLicence(int id)
+        {
+            var licence = licenceRepository.GetById(id);
+
+            return GetLatestStatus(licence).Status.IsLicence ? BuildSummary(licence) : null;
+        }
+
+        private PublicRegisterLicenceSummaryViewModel BuildSummary(Licence licence)
+        {
+            return mapper.Map<PublicRegisterLicenceSummaryViewModel>(licence);
```

Comment:
How does this differ from `BuildLicence` in intended use?

Reply:
This will be changed when the view part is fleshed out. It will change to BuildLicenceDetail or something similar